File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    return queries.map(([s, e, k]) => {
        let list = [];
        for (let i = s; i <= e; i++) {
            if (arr[i] > k) {
                list.push(arr[i]);
            }
        }
        return list.length > 0 ? Math.min(...list) : -1;
    });
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] result = new int[queries.length];
        
        for (int i = 0; i < queries.length; i++) {
            int num = Integer.MAX_VALUE;
            
            for (int idx = queries[i][0]; idx <= queries[i][1]; idx++) {
                if (arr[idx] > queries[i][2] && arr[idx] < num) {
                    num = arr[idx];
                }
            }
            
            if (num == Integer.MAX_VALUE) {
                result[i] = -1;
            } else {
                result[i] = num;
            }
        }
        
        return result;
    }
}